packages/config/src/class-textdomain-customizer.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * The Textdomain_Customizer class.
+ *
+ * @package jetpack-config
+ */
+
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+
+namespace Automattic\Jetpack\Config;
+
+use Composer\Script\Event;
+
+/**
+ * This class is used to customize the textdomain of the strings in the packages.
+ */
+class Textdomain_Customizer {
+
+	/**
+	 * Replaces occurences of the string 'default' (including quotes) with the textdomain
+	 * provided in composer.json. The files that contain translatable strings are listed
+	 * in the $files_to_translate array.
+	 *
+	 * @param Event $event The composer event.
+	 */
+	public static function post_install( Event $event ) {
+
+		$vendor_dir = $event->getComposer()->getConfig()->get( 'vendor-dir' );
+
+		// Do not change the textdomain if this is a development version of the Config package.
+		$installed_packages = json_decode( file_get_contents( $vendor_dir . '/composer/installed.json' ), true );
+		$config_version     = null;
+
+		foreach ( $installed_packages as $package ) {
+			if ( 'automattic/jetpack-config' === $package['name'] ) {
+				$config_version = $package['version'];
+				break;
+			}
+		}
+
+		if ( ! $config_version || 'dev-' === substr( $config_version, 0, 4 ) ) {
+			return;
+		}
+
+		$extra = $event->getComposer()->getPackage()->getExtra();
+
+		if ( ! isset( $extra['textdomain'] ) ) {
+			return;
+		}
+
+		/*
+		 * Array of directories and files that contain translatable strings. The
+		 * textdomain in these files will be set to the plugin's textdomain.
+		 */
+		$items_to_translate = array(
+			'/automattic/jetpack-config/src/example-file.php',
```

Comment:
Can you explain a bit more how you see this working in the end? It's an interesting approach. Are we going to be maintaining this list? We will probably have multiple packages that need to have a dummy textdomain string replaced in their files. Should we have something like an extra field for that file name too?

Reply:
That was my first approach - to maintain that file list with the files that contain strings from all of the packages. However, it's not a great solution.

Your suggestion to use extra fields for the file names led me to take a closer look at the Composer API, and I think I've found a much better approach. I'm working on a new commit that will allow each package to provide its own file list in the extra section of its composer.json file.